keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1199,6 +1200,42 @@ class SqlArtifactRepository(
     }
   }
 
+  override fun getGitMetadataByPromotionStatus(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String,
+    artifactReference: String,
+    promotionStatus: String
+  ): GitMetadata? {
+    return sqlRetry.withRetry(READ) {
+
+      val artifact = deliveryConfig.artifacts.firstOrNull { it.reference == artifactReference }
```

Comment:
These lines here alone would be a good reason to replace the `artifactReference` parameter with `artifact: DeliveryArtifact`. There was not a single occurrence of other repository functions I'v refactored in this way where the call site did not already have an artifact object handy to pass down, and it avoids these kinds of unnecessary lookups.

Reply:
I now see what you are saying @luispollo ! Thanks for pointing it out. Changed to `DeliveryArtifact`.